tools/nodejs/src/statement.cpp

```diff
@@ -34,7 +34,37 @@ struct PrepareTask : public Task {
 
 	void DoWork() override {
 		auto &statement = Get<Statement>();
-		statement.statement = statement.connection_ref->connection->Prepare(statement.sql);
+
+		auto &connection = statement.connection_ref->connection;
+		std::vector<std::unique_ptr<duckdb::SQLStatement>> statements;
+		try {
+			if (connection == nullptr) {
+				throw duckdb::ConnectionException("Connection was never established or has been closed already");
+			}
+			statements = connection->ExtractStatements(statement.sql);
+			if (statements.empty()) {
+				throw duckdb::InvalidInputException("No statement to prepare!");
+			}
+		} catch (const duckdb::Exception &ex) {
+			statement.statement = duckdb::make_unique<duckdb::PreparedStatement>(duckdb::PreservedError(ex));
+			return;
+		} catch (std::exception &ex) {
+			statement.statement = duckdb::make_unique<duckdb::PreparedStatement>(duckdb::PreservedError(ex));
+			return;
+		}
+
+		// if there are multiple statements, we directly execute the statements besides the last one
+		// we only return the result of the last statement to the user, unless one of the previous statements fails
+		for (idx_t i = 0; i + 1 < statements.size(); i++) {
+			auto pending_query = connection->PendingQuery(std::move(statements[i]));
+			auto res = pending_query->Execute();
+
+			if (res->HasError()) {
+				res->ThrowError();
```

Comment:
Should we throw an error here? Could you perhaps try triggering this (e.g. `SELECT * FROM non_existent_table; SELECT 42`)

Reply:
Yea you are right, we should probably do something different than throw a c++ exception here.
I tried to throw a Napi::Error through the Env instead, and also tried just only setting `statement.statement` to a ParsedStatement containing the PreservedError, but in either case it segfaults, instead of resolving properly
```ts
        it('should fail executing the first statement', function(done) {
            db.prepare("select * from non_existent_table; select 42")
                .all(function(err: null | Error, rows: TableData) {
                    if (err) {
						done();
					}
                })
                .finalize(done);
        });
```

Reply:
I looked into it a little more, `.all` runs the `RunPreparedTask` Task, but the `PrepareTask` Task is failing, so likely I need to change this test, `db.prepare()` fails, before `.all()` even runs

Reply:
Should be fixed now